### DISCUSSION TITLE
Autoscaling w.r.t Resource utilization with celery executor

### DISCUSSION THREAD

**Discussion by rahultilokanizs** [apache/airflow#34176] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Hello Team,

We are using Airflow with Celery executor in which we are facing issues with autoscaling, for ex we have a small instance with minimum worker count as 1, so maximum number of tasks that this base worker can get is 5. In our scenario we have 4 task running on the worker and they are already consuming 95 to 100% of the CPU Utilization or Memory Utilization of the worker. If a new task comes to the celery worker it will also be assigned to the same worker even though the worker does not have the resources to cater to the new incoming task. So due to this the task will may be stuck in queued state or get failed with SIGKILL error.

Thanks
Rahul

### What you think should happen instead

We can incorporate autoscaling based on resource utilization, for ex if the base worker hits the 90% mark for a given period of time, Airflow should auto scale and add an additional node to the existing once. So basically combining the current functionality of task based autoscaling with the an additional feature of resource based autoscaling.

Same can be done with downscaling, if the worker goes below a certain percentage of utilization we can downscale the current additional nodes.

### How to reproduce

One can use high CPU and Memory intensive task so that this condition can be replicated.

### Operating System

Amazon Linux

### Versions of Apache Airflow Providers

2.2.2

### Deployment

Amazon (AWS) MWAA

### Anything else

We have been facing this issue from quite some time, and urge the community to have a look on this issue.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
This is a question to MWAA support, not Airflow. The MWAA deployment team manages all MWAA airflow instances and they implement all mechanisms of scaling. You should open a support ticket with them (they have paid support).

**Comment by potiuk** [comment]
Converted to discussion if needed. 

BTW. Airflow already has KEDA autoscaling that is possible to be used via Community Helm Chart. But it's entirely up to the Service providers - in this case MWAA - how they are deploying airflow and which "deployment" capabilities are used from the community (for example autoscaling) is configured/customized - or whether they develop their own solution there).
